index.js

```diff
@@ -8,6 +13,97 @@ server.use(helmet());
 
 // endpoints here
 
+server.get("/", (req, res) => {
+  res.send(
+    "There are three kinds of people in this world, those who can count and those who can not."
+  );
+});
+
+server.get("/api/zoos", (req, res) => {
+  db("zoos")
+    .then(zoos => {
+      res.json(zoos);
+    })
+    .catch(err => {
+      res.status(500).json({
+        error: err,
+        message: "The zoo information could not be retrieved."
+      });
+    });
+});
+
+server.get("/api/zoos/:id", (req, res) => {
+  const zooID = req.params.id;
+  db("zoos")
+    .where({ id: zooID })
+    .then(zoo => {
+      res.status(200).json(zoo);
+    })
+    .catch(err => {
+      res.status(500).json({
+        error: err,
+        message: "The zoo information could not be retrieved."
+      });
+    });
+});
+
+server.post("/api/zoos", (req, res) => {
+  db("zoos")
+    .insert(req.body)
+    .then(zoos => {
+      res.status(201).json({ message: "Successfully created zoo." });
+    })
+    .catch(err => {
+      res.status(500).json({
+        error: err,
+        message: "There was an error while saving the zoo to the database."
+      });
+    });
+});
+
+server.delete("/api/zoos/:id", (req, res) => {
+  const zooID = req.params.id;
+
+  db("zoos")
+    .where({ id: zooID })
+    .delete()
+    .then(zoo => {
+      res.status(201).end();
+    })
+    .catch(err => {
+      res.status(404).json({
+        error: err,
+        message: "The zoo with the specified ID does not exist."
+      });
+    });
+});
+
+server.put("/api/zoos/:id", (req, res) => {
+  db("zoos")
+    .where({ id: req.params.id })
+    .update(req.body)
+    .then(count => {
+      if (count > 0) {
+        db("zoos")
+          .where({ id: req.params.id })
+          .first()
+          .then(zoo => {
+            res.status(200).json({ message: "Successfully updated zoo." });
```

Comment:
Nice work - if you're getting the zoo that's been updated, you might as well send the updated zoo rather than this message. It can make client-side code easier to write.